File: pe/nonvisualstudio/clojure/Implementation/Problem0005.clj
(ns pe.clojure.implementation.problem0005)


(load-file "C:/Users/Thomas/Documents/GitHub/practice/utilities/nonvisualstudio/clojure/Implementation/Math.clj")
(use 'utilities.math)

(defn lcmOfList [upToWhatNumber]
  (reduce lcm (range 1 (+ upToWhatNumber 1))))

File: pe/nonvisualstudio/elixir/Implementation/Problem0005.ex
Code.require_file "../../../../Utilities/nonvisualstudio/elixir/Implementation/Math.ex", __DIR__

defmodule Problem0005 do

  def lcmOfList(upToWhatNumber), do: 2..upToWhatNumber |> Enum.reduce(1, (fn(elem, acc) -> Math.lcm(elem,acc) end))
end
